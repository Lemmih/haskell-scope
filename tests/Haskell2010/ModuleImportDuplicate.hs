-- Make sure importing the same identifiers many times doesn't cause problems.
module ModuleImportDuplicate where

import Stdlib
-- Interesting, importing the same identifier twice does not trigger a warning
-- in ghc-8.2.1, even with -Wall.
import Stdlib (Int, Int)

type MyInt = Int
