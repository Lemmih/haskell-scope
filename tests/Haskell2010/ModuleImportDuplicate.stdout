Scope errors:

Definitions:
  Definition 1, from ModuleImportDuplicate.decl0:
    type [4mMyInt[0m = Int

Use sites:
  Definition used: Stdlib.decl1
    import Stdlib ([4mInt[0m, Int)
  Definition used: Stdlib.decl1
    import Stdlib (Int, [4mInt[0m)
  Definition used: Stdlib.decl1
    type MyInt = [4mInt[0m

Exports:
  MyInt, Type from ModuleImportDuplicate
